core/blockexplorer/src/components/Home/index.js

```diff
@@ -39,14 +40,79 @@ class Home extends Component {
             });
     }
 
+    searchKeyPress = async e => {
+        if(e.keyCode == 13){
+            var searchTerm = e.target.value.trim();
+            
+            if (searchTerm == '') {
+                return;
+            }
+
+            if (this.isBlockNumber(searchTerm)) {
+                this.setState({redirectUrl:'/block/' + searchTerm});
+            } else if (await this.isTransactionHash(searchTerm)) {
+                this.setState({redirectUrl:'/transaction/' + searchTerm});
+            } else if (await this.isAddress(searchTerm)) {
+                this.setState({redirectUrl:'/address/' + searchTerm});
+            }
+            else {
+                alert(searchTerm + ' is not a valid blocknumber or transaction hash');
+            }
+        }
+    }
+
+    isBlockNumber(searchTerm) {
+        if (searchTerm.length > 16) {
+            return false;
+        }
+        const blockNumber = parseInt(searchTerm);
+        console.log('blockNumber', blockNumber);
+        return blockNumber!==0 && blockNumber <= this.state.latestBlock.blockIndex;
+    }
+
+    async isAddress(searchTerm) {
+        try {
+            var response = await fetch(`/api/query/address/${searchTerm}`,{mode: 'cors'});
+            var addressInfo = await response.json();
+
+            if (addressInfo.totalReceived > 0 || addressInfo.totalSent > 0) {
+                return true;
+            }
+
+            // at this point still not sure so do deeper check
```

Comment:
I was thinking about a new address that hasn't been used yet with the below - but actually that is irrelevant isn't it since that means there is no address :)

so maybe just return false instantly here so that we get an "not found" result